loki/transformations/inline.py

```diff
@@ -95,6 +98,8 @@ def __init__(
         self.adjust_imports = adjust_imports
         self.external_only = external_only
         self.resolve_sequence_association = resolve_sequence_association
+        if self.inline_marked:
+            self.creates_items = True
```

Comment:
That pattern is likely my fault, aiming to be conservative in switching this on to avoid the discovery overhead when inlining is limited to member routines only